api/s2n.h

```diff
@@ -98,6 +98,9 @@ extern int s2n_connection_wipe(struct s2n_connection *conn);
 extern int s2n_connection_free(struct s2n_connection *conn);
 extern int s2n_shutdown(struct s2n_connection *conn, s2n_blocked_status *blocked);
 
+typedef enum { S2N_CERT_AUTH_REQUIRED, S2N_CERT_AUTH_NONE } s2n_cert_auth_type;
+extern int s2n_connection_set_client_cert_auth_type(struct s2n_connection *conn, s2n_cert_auth_type cert_auth_type);
+
```

Comment:
Did we want to include `s2n_connection_set_client_cert_verify_callback` in this change? Looks like s2nd is using it but it wasn't added to the api.